azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java

```diff
@@ -152,6 +153,18 @@ public FunctionApp getFunctionApp() throws AzureAuthFailureException {
         return null;
     }
 
+    public RuntimeConfiguration getRuntime() {
+        return runtime;
+    }
+
+    public OperatingSystemEnum getRuntimeOs() throws MojoExecutionException {
```

Comment:
why throw MojoExecutionException  in a get method?

Reply:
We need to parse string to enum, which will throw MojoExecutionException

Reply:
Removed this method and change to get os from `RuntimeConfiguration`.